files/galaxy/tpv/tools.yml

```diff
@@ -25,40 +25,50 @@ tools:
     inherits: basic_gpu_resource_param_tool
 
   toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/.*:
-    # The memory requirement of Hifiasm depends on a wrapper's input
-    mem: |
-      from math import ceil
-
-      parameters = {p.name: p.value for p in job.parameters}
-      parameters = tool.params_from_strings(parameters, app)
-
-      advanced_options = parameters.get("advanced_options", dict())
-
-      kcov_default = 36
-      kcov = advanced_options.get("kcov", kcov_default)
-
-      hg_size = advanced_options.get("hg_size", "")
-
-      value = 0
-      if hg_size:
-          conversion_factors = {
-              "k": 1000000,
-              "M": 1000,
-              "G": 1,
-          }
-          conversion_factors = {
-              key.lower(): value for key, value in conversion_factors.items()
-          }
-          suffix = hg_size[-1:].lower()
-          value = hg_size[:len(hg_size) - 1]
-          value = value.replace(",", ".")
-          value = float(value)
-          # compute hg size in Gb
-          value = value / conversion_factors[suffix.lower()]
-          value = ceil(value * (kcov * 2) * 1.75)
-
-      # return the amount of memory needed
-      value
+    rules:
+      - id: hifiasm_memory
+        # The memory requirement of Hifiasm depends on a wrapper's input
+        if: |
+          parameters = {p.name: p.value for p in job.parameters}
+          parameters = tool.params_from_strings(parameters, app)
+
+          advanced_options = parameters.get("advanced_options", dict())
+          hg_size = advanced_options.get("hg_size", "")
+
+          bool(hg_size)
+        mem: |
+          from math import ceil
+
+          parameters = {p.name: p.value for p in job.parameters}
+          parameters = tool.params_from_strings(parameters, app)
+
+          advanced_options = parameters.get("advanced_options", dict())
+
+          kcov_default = 36
+          kcov = advanced_options.get("kcov", kcov_default)
+
+          hg_size = advanced_options.get("hg_size", "")
+
+          value = 0
```

Comment:
shouldn' that be some sane default value?

Reply:
even if it has no effect because of the if condition ...

Reply:
Typically I would answer yes, but isn't this precisely what let us quickly find out that something was off?

I mean, imagine that we had already set something other than zero here. Then the tool would have been taking mem values different from what is defined in the shared database for a long time and we would have only noticed on very specific tool runs. If it is zero then it fails for all runs and we notice immediately.

Reply:
Good argument.